Firestore/Protos/build_protos.py

```diff
@@ -0,0 +1,204 @@
+#! /usr/bin/python
+
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generates and massages protocol buffer outputs.
+"""
+
+from __future__ import print_function
+
+import sys
+
+import argparse
+import os
+import os.path
+import re
+import shutil
+import subprocess
+import tarfile
+import urllib2
+
+
+NANOPB_VERSION = '0.3.8'
+PROTOC_BIN = 'Pods/!ProtoCompiler/protoc'
+
+COPYRIGHT_NOTICE = '''
+/*
+ * Copyright 2018 Google
```

Comment:
I needed this process to be repeatable without manually frobbing the files.

I'm willing to bet we're not going to add any new proto files soon, so this will actually be correct for the foreseeable future. Pinning this to 2018 for all existing files seems right.

An alternative approach I considered was to run through the contents of the directory before running to collect the prior state of the copyright notice, but I thought that was too much work for now. We can revisit if this starts to be wrong.

I'm unconcerned with someone else using this script for their purposes. This is a implementation detail of our build process. People are welcome to it, but it's not particularly novel or interesting--just glue code that won't directly fit anyone else's needs directly.